### PR TITLE
Move file picker by clicking card into the MediaUploader component

### LINKED ISSUE #35714: [Enhancement]: move file picker by clicking card into the MediaUploader component
### Describe the solution you'd like

Move the logic to open the [file picker by clicking the card](https://github.com/woocommerce/woocommerce/blob/add/35183_open_file_picker_by_clicking_card/plugins/woocommerce-admin/client/products/sections/images-section.tsx#L188-L271) into the [`MediaUploader` component](https://github.com/woocommerce/woocommerce/tree/trunk/packages/js/components/src/media-uploader).

Since all media upload components will have the same behavior of file uploading by clicking the card, we want to move that logic into the component.

More details [here](https://github.com/woocommerce/woocommerce/pull/35358#discussion_r1028155142).

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR moves the code to pick a file after clicking the images card to the MediaUploader component.

Screen Capture on 2022-11-27 at 16-03-29

Closes #35714.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Check out this branch and go to Products > Add New (MVP)
2. Click the _Drag and drop_ area (not the `Choose images` button).
3. A file picker should be visible. Pick an image.
4. The selected image should be visible now in the gallery.
5. The _tracks_ event `wcadmin_product_images_add_via_file_upload_area` should be recorded
6. Continue with the product creation process. Confirm that it has been created correctly.

**Storybook**
Check that the storybook for the component `MediaUploader` is working as expected.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/sections/images-section.tsx [hunk 1]
```diff
@@ -185,89 +179,50 @@ export const ImagesSection: React.FC = () => {
...
-																'woocommerce'
-															) }
-														</span>
-													</>
-												}
+									label={
+										<>
+											<img
+												src={ DragAndDrop }
+												alt="Completed"
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/products/sections/images-section.tsx]: I know this was from the original changes, but should this be wrapped in `__(`?

**Review by louwie17 — DISMISSED** [review]
This tested well, nice work @octaedro ! 💯

File: plugins/woocommerce-admin/client/products/sections/images-section.tsx (hunk 1 above)
Comment (octaedro) [reply plugins/woocommerce-admin/client/products/sections/images-section.tsx]: Nice catch @louwie17!
I just pushed this fix. Could you take another quick look at this PR?

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 Thanks for updating the completed text 😄

**Pull request merged**